### PR TITLE
Specialist lesson listing LESQ-501

### PR DESCRIPTION
**Pull request by k-huggs**
## Description

Music year: 1988

- Creates Specialist Lesson Listing view, fixture, tests and story

## Issue(s)

Fixes #LESQ-501

## How to test

1. Go to https://deploy-preview-2199--owa-storybook.netlify.thenational.academy/?path=/story/components-teacherviews-specialistlessonlisting--specialist-lesson-listing-page

## Screenshots

How it sho
Screenshot 2024-01-18 at 14 06 25
uld now look:

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

**Review by jack-skerrett-oak — APPROVED** [review]
It looks like a lesson listing page to me

**Review by Monika-Koziol — APPROVED** [review]
Looks fine, it shouldn't have the new tag as it's old content.

File: src/components/TeacherComponents/ListItemHeader/ListItemHeader.tsx [hunk 1]
```diff
@@ -22,13 +23,15 @@ interface CommonProps {
...
+  onClick: () => void;
 }
 
-type SpecialistListItemProps = IndividualSpecialistUnit &
+type SpecialistListItemProps = (IndividualSpecialistUnit | SpecialistLesson) &
   CommonProps & {
-    title: LessonListItemProps["lessonTitle"] | UnitListItemProps["title"];
-    slug: LessonListItemProps["lessonSlug"] | UnitListItemProps["slug"];
+    title: SpecialistLesson["lessonTitle"] | IndividualSpecialistUnit["title"];
+    slug: SpecialistLesson["lessonSlug"] | IndividualSpecialistUnit["slug"];
```
Comment (njblore) [inline src/components/TeacherComponents/ListItemHeader/ListItemHeader.tsx:32]: I know you didn't add this here, but these are just strings right? I'm not sure the extra typing is helpful 😅

File: src/components/TeacherViews/SpecialistLessonListing/SpecialistLessonListing.view.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        <Grid data-testid="specialist-lesson-grid">
+          <GridArea $colSpan={[12, 9]} $mt={[16, 32]}>
+            <LessonList
+              {...curriculumData}
+              lessonCount={lessons.length}
+              currentPageItems={currentPageItems}
+              paginationProps={paginationProps}
+              headingTag={"h2"}
+              unitTitle={unitTitle}
+              onClick={() => {}}
```
Comment (njblore) [inline src/components/TeacherViews/SpecialistLessonListing/SpecialistLessonListing.view.tsx:108]: should this be doing something?

File: src/components/TeacherComponents/ListItemHeader/ListItemHeader.tsx (hunk 1 above)
Comment (k-huggs) [reply src/components/TeacherComponents/ListItemHeader/ListItemHeader.tsx:32]: ahaha yep, changed✅

File: src/components/TeacherViews/SpecialistLessonListing/SpecialistLessonListing.view.tsx (hunk 2 above)
Comment (k-huggs) [reply src/components/TeacherViews/SpecialistLessonListing/SpecialistLessonListing.view.tsx:108]: We don't have the tracking set up yet for the on clicks, I've dropped a message to the data team to up date

Comment (njblore) [reply src/components/TeacherViews/SpecialistLessonListing/SpecialistLessonListing.view.tsx:108]: aha cool! maybe add a TODO in so the next person to see this isn't confused

**Review by njblore — APPROVED** [review]

**Pull request merged**